### PR TITLE
New randomizer

### PR DESCRIPTION
**Pull request by OsnaFonz**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1658,14 +1700,27 @@
...
+		if(num == undefined) num = arr.length;
+		
+	/*
 		for(var i = 0; i < num; i++) {
 			var j = Math.floor(Math.random() * (arr.length - i - 1)) + i + 1;
 			var temp = arr[i];
 			arr[i] = arr[j];
 			arr[j] = temp;
 		}
+		*/
```
Comment (Arachnid) [inline index.html:1712]: Why the change to the shuffling function? The old function only shuffles as many elements as are required to return the requested result set. And why leave the old function here but commented out?

**Comment by Arachnid** [comment]
Nice, I like it!

**Comment by OsnaFonz** [comment]
The old routine didn't worked with a three item array.

File: index.html (hunk 1 above)
Comment (OsnaFonz) [inline index.html:1712]: The old routine didn't worked with a three item array. (i removed it now)

Comment (Arachnid) [inline index.html]: It looks like it was a simple off-by-1 error, easy enough to correct without rewriting everything!
